tracing-core/src/dispatcher.rs

```diff
@@ -392,7 +473,7 @@ impl Dispatch {
     #[inline]
     pub fn none() -> Self {
         Dispatch {
-            subscriber: Arc::new(NoSubscriber),
+            subscriber: Arc::new(Inner(NoSubscriber)),
```

Comment:
Let's not give `NoSubscriber` an `Inner`. `NoSubscriber`'s interests will never be cached, and new instances of it are created or destroyed frequently, so it shouldn't blow away the cache:
```suggestion
            subscriber: Arc::new(NoSubscriber),
```